### ISSUE TITLE
Wrong type detection

### ISSUE THREAD

**Issue by Migoyan** [PyCQA/pylint#7704]
### Bug description

```py
import numpy as np

my_array = np.array([[0,1,np.nan], [np.nan,4,np.nan]])
my_masked_array = np.ma.masked_invalid(my_array)

my_masked_array.compressed() # pylint report an error: Instance of 'ndarray' has no 'compressed' member
```

pylint detect `my_masked_array` type as an `ndarray` although `np.ma.masked_invalid` return a `MaskedArray`.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:8:6: E1101: Instance of 'ndarray' has no 'compressed' member (no-member)
```

### Expected behavior

pylint should properly detect the type `MaskedArray` and not report an error.

### Pylint version

```shell
python 3.7.3
pylint 2.15.5
numpy 1.21.6
```

### OS / Environment

Debian 10

**Comment by mbyrnepr2** [comment]
Thanks for reporting this @Migoyan. This is so similar to https://github.com/PyCQA/pylint/issues/5715 that I think we should close this one as a duplicate.
